samples/sample_multi_transcode/src/pipeline_transcode.cpp

```diff
@@ -4499,27 +4420,31 @@ mfxStatus CTranscodingPipeline::Reset()
 mfxStatus CTranscodingPipeline::AllocAndInitVppDoNotUse(sInputParams *pInParams)
 {
     std::vector<mfxU32> filtersDisabled;
-    if(pInParams->DenoiseLevel==-1)
+    if (pInParams->DenoiseLevel == -1)
     {
-        filtersDisabled.push_back(MFX_EXTBUFF_VPP_DENOISE); // turn off denoising (on by default)
+        filtersDisabled.emplace_back(MFX_EXTBUFF_VPP_DENOISE); // turn off denoising (on by default)
     }
-    filtersDisabled.push_back(MFX_EXTBUFF_VPP_SCENE_ANALYSIS); // turn off scene analysis (on by default)
+    filtersDisabled.emplace_back(MFX_EXTBUFF_VPP_SCENE_ANALYSIS); // turn off scene analysis (on by default)
 
-    m_VppDoNotUse.NumAlg = (mfxU32)filtersDisabled.size();
+    if (filtersDisabled.size())
+    {
+        auto doNotUse = m_mfxVppParams.AddExtBuffer<mfxExtVPPDoNotUse>();
+        doNotUse->NumAlg = (mfxU32)filtersDisabled.size();
 
-    if(m_VppDoNotUse.AlgList)
-        delete[] m_VppDoNotUse.AlgList;
-    m_VppDoNotUse.AlgList = new mfxU32 [m_VppDoNotUse.NumAlg];
-    MSDK_CHECK_POINTER(m_VppDoNotUse.AlgList,  MFX_ERR_MEMORY_ALLOC);
-    MSDK_MEMCPY(m_VppDoNotUse.AlgList,&filtersDisabled[0],sizeof(mfxU32)*filtersDisabled.size());
+        if (doNotUse->AlgList)
```

Comment:
deleting ```nullptr``` is valid call, it just does nothing. So this check can be removed

Reply:
ok